### PR TITLE
Feature/39 implement cqrs delivery service

### LINKED ISSUE #39: implement CQRS
**Description**
implement CQRS for Delivery service

**Tasks**
- [ ] ...

### PR DESCRIPTION
**Pull request by arjennorbart**
## Changes
A short description about the overall changes you've made to the project.

## Checklist
- [x] Implement Command and Query handler
- [x] Add tests
- [x] Format all modified files
- [x] Check for missing newlines

## When merging, this pull request 
closes #39

### CODE REVIEW

**Review by JustMilan — APPROVED** [review]

File: delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/application/DeliveryQueryHandler.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@Service
+public class DeliveryQueryHandler {
+    private final DeliveryRepository repository;
+
+    public DeliveryQueryHandler(DeliveryRepository repository) {
+        this.repository = repository;
+    }
+
+    public Delivery handleGetDeliveryById(UUID id) {
+        return this.repository.findById(id).orElseThrow();
```
Comment (xandervedder) [inline delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/application/DeliveryQueryHandler.java:20]: nit: Custom Exception?

**Review by xandervedder — CHANGES_REQUESTED** [review]

File: delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/domain/Delivery.java [hunk 2]
```diff
@@ -72,4 +91,12 @@ public Status getStatus() {
     public UUID getOrderId() {
         return orderId;
     }
+
+    public void clearEvents() {
+        this.events.clear();
+    }
+
+    public List<DeliveryEvent> listEvents() {
+        return this.events;
```
[the remark this answers is not in the document]
Comment (arjennorbart) [reply delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/domain/Delivery.java:100]: If i do this the event list cannot be cleared with the clearEvents() method

[the remark this answers is not in the document]
Comment (arjennorbart) [reply delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/domain/Delivery.java]: fixed

File: delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/application/DeliveryCommandHandler.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        delivery.changeAddress(address);
+        this.publishEventsFor(delivery);
+        this.repository.save(delivery);
+    }
+
+    // Will eventually be used by the Controller
+    public Delivery handleRegisterDelivery(DeliveryAddress address, UUID orderId) {
+        var delivery = new Delivery(address, Status.REGISTERED, orderId);
+        this.publishEventsFor(delivery);
+        this.repository.save(delivery);
```
[the remark this answers is not in the document]
Comment (arjennorbart) [reply delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/application/DeliveryCommandHandler.java]: fixed

**Review by xandervedder — APPROVED** [review]

**Pull request merged**